### PR TITLE
Ajax aka not the "proto" anymore

### PR DESCRIPTION
**Pull request by MaxShehovtsov**
Let's give it a shot :beers:

### CODE REVIEW

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+﻿/// <reference path="../../testframework/common.ts" />
+/// <reference path="../../testframework/contracttesthelper.ts" />
+/// <reference path="../../../JavaScriptSDK/telemetry/RemoteDependencyData.ts" />
+/// <reference path="../../../JavaScriptSDK/Contracts/Generated/SeverityLevel.ts" />
+class RemoteDependencyTests extends ContractTestHelper {
+
+    private exception;
+    private static url = "http://myurl.com"
+    private static isAsync = true;
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts]: we never use ```isAsync``` in UI. Let's not have it mentioned in JavaScript

File: JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            name: false,
+            kind: true,
+            value: false,
+            count: false,
+            min: false,
+            max: false,
+            stdDev: false,
+            dependencyKind: false,
+            success: false,
+            async: false,
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts]: again, ```async``` it is nullable. Let's not have it in JavaScript code

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts (hunk 1 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts]: I think you can see it in dependency's properties "..."

File: JavaScript/JavaScriptSDK/ajax/ajax.ts [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+                                    this.ajaxData.requestSize += content.length;
+                                }
+                            }
+
+                            this.ajaxData.requestSentTime = dateTime.Now();
+                            this.ajaxData.loadingRequest = document.readyState === "loading";
+
+                            if (!this.ajaxData.onreadystatechangeCallbackAttached) {
+
+                                // IE 8 and below does not support xmlh.addEventListener. This the last place for the browsers that does not support addEventListenener to intercept onreadystatechange
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK/ajax/ajax.ts]: Do we support IE 8 at all? Maybe we shouldn't even intercept XmlHttpRequest for older browsers?

File: JavaScript/JavaScriptSDK/ajax/ajax.ts [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+
+                function onAjaxComplete() {
+                    commands.TryCatchTraceWrapper.call(this, "publishData", function () {
+                        this.ajaxData.CalculateMetrics();
+
+                        appInsights.trackAjax(
+                            this.ajaxData.getAbsoluteUrl(),
+                            this.ajaxData.async,
+                            this.ajaxData.ajaxTotalDuration,
+                            this.ajaxData.status == "200"
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK/ajax/ajax.ts]: why "aborted" status is not reflected as failed request?

File: JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+module Microsoft.ApplicationInsights.Telemetry {
+    "use strict";
+
+    export class RemoteDependencyData extends AI.RemoteDependencyData implements ISerializable {
+
+        public static envelopeType = "Microsoft.ApplicationInsights.RemoteDependencyData";
+        public static dataType = "RemoteDependencyData";
+
+        public aiDataContract = {
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts]: How is this class used?

**Comment by SergeyKanzhelev** [comment]
I cannot find where you enable correlation by ```parentId```

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts (hunk 1 above)
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts]: Let's remove it anyway. I'll remove it from server SDK as well.

File: JavaScript/JavaScriptSDK/snippet.js [hunk 6]
```diff
@@ -30,7 +30,7 @@
         }
     };
 
-    var method = ["Event", "Exception", "Metric", "PageView", "Trace"];
+    var method = ["Event", "Exception", "Metric", "PageView", "Trace", "Ajax"];
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK/snippet.js]: should we notify Arun to update the version of JS snippet for Visual Studio 2015 Update 1?

File: JavaScript/JavaScriptSDK/ajax/ajax.ts (hunk 4 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/ajax/ajax.ts]: I'm not sure it's considered failed, if you manually abort it with xhr.abort()

File: JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts (hunk 5 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts]: This is the data class I construct inside appInsights.trackAjax() - https://github.com/Microsoft/ApplicationInsights-JS/blob/ajax_proto/JavaScript/JavaScriptSDK/appInsights.ts#L282

File: JavaScript/JavaScriptSDK/snippet.js (hunk 6 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/snippet.js]: probably. Thanks!

File: JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts (hunk 2 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts]: Is default value true on backend?

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts (hunk 1 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts]: What about sync ajaxes? https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest#open()

**Pull request merged**